lti_consumer/api.py

```diff
@@ -34,15 +38,26 @@ def _get_or_create_local_lti_config(lti_version, block_location, external_id=Non
     # The create operation is only performed when there is no existing configuration for the block
     lti_config, _ = LtiConfiguration.objects.get_or_create(location=block_location)
 
-    if lti_config.version != lti_version:
-        lti_config.version = lti_version
-
-    if external_id:
-        lti_config.config_store = LtiConfiguration.CONFIG_EXTERNAL
-        lti_config.external_id = external_id
+    config_store = None
+    if config_type == "external" or external_id is not None:
+        config_store = LtiConfiguration.CONFIG_EXTERNAL
+    elif config_type == "database":
+        config_store = LtiConfiguration.CONFIG_ON_DB
     else:
-        lti_config.config_store = LtiConfiguration.CONFIG_ON_XBLOCK
-        lti_config.external_id = None
+        config_store = LtiConfiguration.CONFIG_ON_XBLOCK
```

Comment:
config_type new is the default but it ends up with config store xblock - could we just call config_type as config_store instead and the default is "xblock" not "new"?

it would also be nice to have this be else if type = "new" (or "xblock" if you like the above) and the final default else be else: throw exception "wtf is this that is not a type"